internal/kaas/nodeGroup.go

```diff
@@ -0,0 +1,252 @@
+	}
+
+	return nodeGroups, nil
+}
+
+// GetNodeGroupListConfig returns the machinePool or machineDeployment configurations used by each nodeGroup.
+func GetNodeGroupListConfig(k *k8s.Kubernetes, clusterName string) ([]*NodeGroup, error) {
+
+	var nodeGroups []*NodeGroup
+	var validationErr error
+
+	nodePoolErr := map[string]error{
+		"machineDeploymentErr": nil,
+		"machinePoolErr":       nil,
+	}
+
+	// Check if is machinePool
+	machinePools, machinePoolErr := k.ListMachinePool(clusterName)
+	if machinePoolErr != nil {
+		clientErr, ok := machinePoolErr.(*clientError.ClientError)
+		if !ok {
+			nodePoolErr["machinePoolErr"] = clientError.NewClientError(machinePoolErr, clientError.UnexpectedError, fmt.Sprintf("Error while listing machinePool for all nodegroups of the cluster %s", clusterName))
+		} else {
+			if clientErr.ErrorMessage != clientError.EmptyResponse {
+				nodePoolErr["machinePoolErr"] = clientError.NewClientError(clientErr, clientError.UnexpectedError, fmt.Sprintf("Error while listing machinePool for all nodegroups of the cluster %s", clusterName))
+			}
+		}
+	} else {
+		if len(machinePools.Items) != 0 {
+			for _, machinePool := range machinePools.Items {
+				validationErr = k8s.ValidateMachineTemplateComponents(machinePool.Spec.Template)
+				if validationErr != nil {
+					log.Printf("skipping invalid MachinePool %s: %s", machinePool.Name, validationErr.Error())
+					continue
+				}
+				nodeGroup := &NodeGroup{
+					Name:               GetNodeGroupShortName(machinePool.Spec.ClusterName, machinePool.Name),
+					Cluster:            machinePool.Spec.ClusterName,
+					InfrastructureKind: machinePool.Spec.Template.Spec.InfrastructureRef.Kind,
+					InfrastructureName: machinePool.Spec.Template.Spec.InfrastructureRef.Name,
+					Replicas:           machinePool.Spec.Replicas,
+				}
+				nodeGroups = append(nodeGroups, nodeGroup)
+			}
+
+			if len(nodeGroups) == 0 {
+				return nil, clientError.NewClientError(validationErr, clientError.EmptyResponse, fmt.Sprintf("No valid NodeGroups were found in the cluster %v, some Nodegroups have invalid configuration", clusterName))
+			}
+			if nodePoolErr["machinePoolErr"] == nil {
+				return nodeGroups, nil
+			}
+		}
+	}
+
+	machineDeployments, machineDeploymentErr := k.ListMachineDeployment(clusterName)
+	if machineDeploymentErr != nil {
+		clientErr, ok := machineDeploymentErr.(*clientError.ClientError)
+		if !ok {
+			nodePoolErr["machineDeploymentErr"] = clientError.NewClientError(clientErr, clientError.UnexpectedError, fmt.Sprintf("Error while listing machineDeployment for all nodegroups of the cluster %s", clusterName))
+		} else {
+			if clientErr.ErrorMessage != clientError.EmptyResponse {
+				nodePoolErr["machineDeploymentErr"] = clientError.NewClientError(machineDeploymentErr, clientErr.ErrorMessage, fmt.Sprintf("Error while listing machineDeployment for all nodegroups of the cluster %s", clusterName))
```

Comment:
`machineDeployment` should be `MachineDeployment`
`nodegroups` should be `NodeGroups`